Client/core/CChat.cpp

```diff
@@ -699,6 +736,46 @@ bool CChat::CharacterKeyHandler(CGUIKeyEventArgs KeyboardArgs)
     return true;
 }
 
+bool CChat::KeyDownHandler(CGUIKeyEventArgs KeyboardArgs)
+{
+    // If we can take input
+    if (!CLocalGUI::GetSingleton().GetConsole()->IsVisible() && IsInputVisible())
+    {
+        switch (KeyboardArgs.scancode)
+        {
+            case CGUIKeys::Scan::ArrowUp:
+            {
+                // If there's nothing to select, break here
+                if (m_InputHistory.vecEntryList.size() <= 1 || m_uiSelectedInputHistoryEntry == 1)

```

Comment:
So reading this code, is the logic like this:

`m_InputHistory = ["currentEmpty", "oldest", "old", "most recent"]`

So 0 is a special case, and the more recent the value, the larger the entry number?

Reply:
Correct. I found this convenient, although it might make more sense to keep the current one always last so order is always one direction without this special case.

Reply:
This has now been changed so that history never stores the current input as it isn't officially part of the history until it has been submitted. I made a new member for both CChat and CConsole that contains the temporary input string. I believe it is cleaner this way.